crates/store/src/genesis/config/mod.rs

```diff
@@ -0,0 +1,333 @@
+
+            faucets.insert(symbol.clone(), faucet_account.id());
+
+            secrets.push((
+                format!("faucet_{symbol}.mac", symbol = symbol.to_lowercase()),
+                faucet_account.clone(),
+                secret_key,
+                faucet_account_seed,
+            ));
+
+            all_accounts.push(faucet_account);
+        }
+
+        // then setup all wallet accounts, which reference the faucet's for their provided assets
+        for (index, WalletConfig { is_updatable, storage_mode, assets }) in
+            wallet_configs.into_iter().enumerate()
+        {
+            let mut rng = ChaCha20Rng::from_seed(rand::random());
+            let secret_key = SecretKey::with_rng(&mut get_rpo_random_coin(&mut rng));
+            let auth = AuthScheme::RpoFalcon512 { pub_key: secret_key.public_key() };
+            let init_seed: [u8; 32] = rng.random();
+
+            let assets = Result::<Vec<_>, Error>::from_iter(assets.into_iter().map(
+                |AssetEntry { amount, symbol }: AssetEntry| {
+                    let token_symbol = TokenSymbol::new(&symbol)?;
+                    let faucet_id = faucets
+                        .get(&symbol)
+                        .ok_or_else(|| Error::MissingFaucetDefinition { symbol: token_symbol })?;
+
+                    Ok(FungibleAsset::new(*faucet_id, amount)?)
+                },
+            ))?;
+            let account_type = if is_updatable {
+                AccountType::RegularAccountUpdatableCode
+            } else {
+                AccountType::RegularAccountImmutableCode
+            };
+            let account_storage_mode = storage_mode.into();
+            let (mut wallet_account, wallet_account_seed) =
+                create_basic_wallet(init_seed, auth, account_type, account_storage_mode)?;
+
+            // Add fungible assets.
+            let mut fungible_assets = FungibleAssetDelta::default();
+            assets
+                .into_iter()
+                .try_for_each(|fungible_asset| fungible_assets.add(fungible_asset))?;
+
+            // Force the account nonce to 1.
+            //
+            // By convention, a nonce of zero indicates a freshly generated local account that has
+            // yet to be deployed. An account is deployed onchain along with its first
+            // transaction which results in a non-zero nonce onchain.
+            //
+            // The genesis block is special in that accounts are "deplyed" without transactions and
+            // therefore we need bump the nonce manually to uphold this invariant.
+            let delta = AccountDelta::new(
+                wallet_account.id(),
+                AccountStorageDelta::default(),
+                AccountVaultDelta::new(fungible_assets, NonFungibleAssetDelta::default()),
+                Felt::ONE,
+            )?;
+            wallet_account.apply_delta(&delta)?;
+
+            secrets.push((
+                format!("wallet_{index:02}.mac"),
```

Comment:
This might cause issues for programmatically handling this when we have more than 100 wallets.

I do like that it lets the filesystem order them correctly, but the trade-off is that users can't do more than 100 without _knowing_ that this is a special number.


```suggestion
                format!("wallet_{index}.mac"),
```

Reply:
At generation time we actually would know the number, and could use the syntax `{index:0width$}` where `let width = usize::ilog(n);`, it just seemed unrealistic to have more than 100 keys. Will add

Reply:
I'm more thinking from the user's perspective. But maybe I'm over-thinking it.

Essentially you can't write user code that generates `n` files and then does something with `n` files because file `i`'s name depend on the size of `log10(n)`.

Reply:
tl;dr I think sorting is preferable


Usecases:
1. user wants to read/ingest all keys
2. user wants to read a specifc key
3. user wants to know the length of all keys

---
1. in this case the user already needs to know the index, which means leakage of information from the toml into the user code, so having `n` isn't a signficant requirement
2. the user wants to read all key files - in case of consecutive numbers indeed we have predictable paths and can use that to our advantage, for `n`-derived/`0`-prefixed names we have to find the first via `glob` or `regex` in the directory listing and derive the zero padding length. Then we can do the same. This is an additional step. Alternatively one can use the dirlisting and use a `regex` match `account0*[0-9]+` to find all keys. 
3. either way the user either needs the information hardcoded or use the directory listing with a regex pattern, or test all key indices consecutively (or via binary expansion search) - only relevant for large key sizes

Best case, we extract the generation as a library and the user would use the API return value, in which case it doesn't matter and things happen in memory, which is my preferred path.
For file based keys, I'd prefer to use an optional path to allow the user to name them as desired.